app/src/main/java/openfoodfacts/github/scrachx/openfood/views/adapters/HistoryListAdapter.java

```diff
@@ -48,7 +48,7 @@ public void onBindViewHolder(HistoryScanHolder holder, int position) {
         Date date = list.get(position).getTime();
 
         long time = date.getTime();
-        String timeStamp = (String)DateUtils.getRelativeTimeSpanString(time);
+        String timeStamp = String.valueOf(DateUtils.getRelativeTimeSpanString(time,System.currentTimeMillis(), DateUtils.HOUR_IN_MILLIS,DateUtils.FORMAT_ABBREV_RELATIVE));
```

Comment:
Timestamps within the last hour are rounded to "0 hr. ago". I personally prefer showing "X min. ago" for those cases and "A moment ago" or "Just now" for timestamps within the last minute.

Reply:
[DateUtils]( https://developer.android.com/reference/android/text/format/DateUtils.html) does not provide that functionality.

Reply:
We may have to use some library.